pkg/tfvars/openstack/openstack.go

```diff
@@ -60,11 +62,27 @@ func TFVars(masterConfig *v1alpha1.OpenstackProviderSpec, cloud string, external
 	cfg.BaseImageName = imageName
 	if isURL {
 		// Valid URL -> use baseImage as a URL that will be used to create new Glance image with name "<infraID>-rhcos".
-		localFilePath, err := cache.DownloadImageFile(baseImage)
+		var localFilePath string
+
+		url, err := url.Parse(baseImage)
 		if err != nil {
 			return nil, err
 		}
 
+		// We support 'http(s)' and 'file' schemes. If the scheme is http(s), then we will upload a file from that
+		// location. Otherwise will take local file path from the URL.
+		switch url.Scheme {
+		case "http", "https":
+			localFilePath, err = cache.DownloadImageFile(baseImage)
+			if err != nil {
+				return nil, err
+			}
+		case "file":
+			localFilePath = filepath.FromSlash(url.Path)
+		default:
```

Comment:
Personally I would have liked it to be part of validation of the install-config.yaml

But this is fine too.

/approve